docs/explanation/e-cryptography.md

```diff
@@ -0,0 +1,78 @@
+# Cryptography
+
+## Resource checksums
+
+Every version of the Charmed Kafka and Charmed ZooKeeper operators install a pinned revision of the Charmed Kafka snap
+and Charmed ZooKeeper, respectively, in order to 
+provide reproducible and secure environments. The Charmed Kafka snap and Charmed ZooKeeper snap package the 
+Kafka and ZooKeeper workload together with 
+a set of dependencies and utilities required by the lifecycle of the operators. 
+Every artifact bundled into the Charmed Kafka snap and Charmed ZooKeeper snap are verified against their checksum. 
+
+## Sources verification
+
+Charmed Kafka and Charmed ZooKeeper artifacts are published programmatically using release pipelines implemented via GitHub Actions of the 
+associated repositories. 
+All repositories containing the source code of the Charmed Kafka project (i.e. charm and snap codebases) are set up with 
+
+* Branch protection rules requiring additional commits to be merged via Pull-Request with at least 2 approvals from repository maintainers
+* Signed commits requirements (using GPG keys)
+* Requiring developers to sign CLA agreement
```

Comment:
```suggestion
* new commits to be merged via Pull-Request with at least 2 approvals from repository maintainers
* new commits to be signed using GPG keys
* developers to sign a Contributor License Agreement (CLA)
```

Can we provide a link here to one such agreement as an example?

Reply:
added the link